src/main/java/com/google/api/generator/gapic/model/Message.java

```diff
@@ -71,7 +73,22 @@ public boolean hasResource() {
 
   /** Returns the first list repeated field in a message, unwrapped from its list type. */
   @Nullable
-  public Field findAndUnwrapFirstRepeatedField() {
+  public Field findAndUnwrapPaginatedRepeatedField() {
+    for (Field field : fields()) {
+      if (field.isMap()) {
+        List<Reference> repeatedGenericMapRefs = field.type().reference().generics();
+
+        TypeNode paginateType =
```

Comment:
Nit: Did you mean `paginatedType` or `paginationType`?

Reply:
Yeah, a typo, I think I meant `paginatedType`, renamed.